fastcrypto-vdf/src/vdf/pietrzak/mod.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::math::parameterized_group::{multiply, ParameterizedGroupElement};
+use crate::vdf::VDF;
+use fastcrypto::error::FastCryptoError::{InvalidInput, InvalidProof};
+use fastcrypto::error::FastCryptoResult;
+use fastcrypto::hash::{HashFunction, Keccak256};
+use num_bigint::BigUint;
+use num_integer::Integer;
+use serde::Serialize;
+use std::mem;
+
+/// Default size in bytes of the Fiat-Shamir challenge used in proving and verification.
+pub const DEFAULT_CHALLENGE_SIZE_IN_BYTES: usize = 32;
```

Comment:
please document here the soundness of the protocol (i.e., the interactive public coin one) and why 32 bytes is enough here